java/integTest/src/test/java/glide/standalone/CommandTests.java

```diff
@@ -364,4 +359,74 @@ public void functionLoad() {
         assertEquals(libName, regularClient.functionLoadReplace(newCode).get());
         // TODO test with FCALL
     }
+
+    @Test
+    @SneakyThrows
+    public void functionStats_and_functionKill() {
+        assumeTrue(REDIS_VERSION.isGreaterThanOrEqualTo("7.0.0"), "This feature added in redis 7");
+
+        String libName = "functionStats_and_functionKill";
+        String funcName = "deadlock";
+        String code = createLuaLibWithLongRunningFunction(libName, funcName, 15);
+        String error = "";
+
+        try {
+            // nothing to kill
+            var exception =
+                    assertThrows(ExecutionException.class, () -> regularClient.functionKill().get());
+            assertInstanceOf(RequestException.class, exception.getCause());
+            assertTrue(exception.getMessage().toLowerCase().contains("notbusy"));
+
+            // load the lib
+            assertEquals(libName, regularClient.functionLoadReplace(code).get());
+
+            try (var testClient =
+                    RedisClient.CreateClient(commonClientConfig().requestTimeout(7000).build()).get()) {
+                // call the function without await
+                // TODO use FCALL
+                var promise = testClient.customCommand(new String[] {"FCALL", funcName, "0"});
+
+                int timeout = 5200; // ms
+                while (timeout > 0) {
+                    var response = regularClient.customCommand(new String[] {"FUNCTION", "STATS"}).get();
+                    if (((Map<String, Object>) response).get("running_script") != null) {
+                        break;
+                    }
+                    Thread.sleep(100);
+                    timeout -= 100;
+                }
+                if (timeout == 0) {
+                    error += "Can't find a running function.";
```

Comment:
I wonder if we can just fail and jump to the finally block here? It seems strange to continue with the test even though something has gone wrong

Reply:
I agree. I give a chance to the test to kill the function (maybe it is running even it can't be found) and do other checks. I raise this error as test failure to ensure it is seen.